toolchain/check/testdata/struct/fail_duplicate_name.carbon

```diff
@@ -8,48 +8,48 @@
 // TIP: To dump output, run:
 // TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/struct/fail_duplicate_name.carbon
 
-// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+7]]:42: error: duplicated field name `abc` in struct type literal [StructNameDuplicate]
+// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+7]]:21: error: duplicated field name `abc` in struct type literal [StructNameDuplicate]
 // CHECK:STDERR: fn F() -> {.d: i32, .abc: i32, .e: i32, .abc: i32, .f: i32};
-// CHECK:STDERR:                                          ^~~
-// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+4]]:22: note: field with the same name here [StructNamePrevious]
+// CHECK:STDERR:                     ^~~~~~~~~
+// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+4]]:41: note: field with the same name here [StructNamePrevious]
 // CHECK:STDERR: fn F() -> {.d: i32, .abc: i32, .e: i32, .abc: i32, .f: i32};
-// CHECK:STDERR:                      ^~~
+// CHECK:STDERR:                                         ^~~~~~~~~
 // CHECK:STDERR:
 fn F() -> {.d: i32, .abc: i32, .e: i32, .abc: i32, .f: i32};
 
-// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+7]]:19: error: duplicated field name `a` in struct type literal [StructNameDuplicate]
+// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+7]]:9: error: duplicated field name `a` in struct type literal [StructNameDuplicate]
 // CHECK:STDERR: let v: {.a: i32, .a: i32} = {.a = 1};
-// CHECK:STDERR:                   ^
-// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+4]]:10: note: field with the same name here [StructNamePrevious]
+// CHECK:STDERR:         ^~~~~~~
+// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+4]]:18: note: field with the same name here [StructNamePrevious]
 // CHECK:STDERR: let v: {.a: i32, .a: i32} = {.a = 1};
-// CHECK:STDERR:          ^
+// CHECK:STDERR:                  ^~~~~~~
 // CHECK:STDERR:
 let v: {.a: i32, .a: i32} = {.a = 1};
 
-// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+7]]:26: error: duplicated field name `def` in struct literal [StructNameDuplicate]
+// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+7]]:32: error: duplicated field name `def` in struct literal [StructNameDuplicate]
 // CHECK:STDERR: let w: i32 = {.def = 1, .def = 2}.def;
-// CHECK:STDERR:                          ^~~
-// CHECK:STDERR: fail_duplicate_name.carbon:[[@LINE+4]]:16: note: field with the same name here [StructNamePrevious]
+// CHECK:STDERR:                                ^
```

Comment:
This location information doesn't seem to be as good as before.

Reply:
Fixed, should be back to what it was. Probably a better approach for transferring names too.